engine/crates/wasi-component-loader/src/context.rs

```diff
@@ -1,23 +1,85 @@
 use std::{collections::HashMap, sync::Arc};
 
+use crossbeam::{channel::TrySendError, sync::WaitGroup};
+use grafbase_telemetry::span::GRAFBASE_TARGET;
 use wasmtime::{
-    component::{LinkerInstance, Resource, ResourceType},
+    component::{ComponentType, LinkerInstance, Lower, Resource, ResourceType},
     StoreContextMut,
 };
 
 use crate::{
     names::{
-        CONTEXT_DELETE_METHOD, CONTEXT_GET_METHOD, CONTEXT_RESOURCE, CONTEXT_SET_METHOD, SHARED_CONTEXT_GET_METHOD,
-        SHARED_CONTEXT_RESOURCE,
+        CONTEXT_DELETE_METHOD, CONTEXT_GET_METHOD, CONTEXT_RESOURCE, CONTEXT_SET_METHOD,
+        SHARED_CONTEXT_ACCESS_LOG_METHOD, SHARED_CONTEXT_GET_METHOD, SHARED_CONTEXT_RESOURCE,
     },
     state::WasiState,
 };
 
+/// Sender for a wasi hook to send logs to the writer.
+pub type ChannelLogSender = crossbeam::channel::Sender<AccessLogMessage>;
+
+/// A receiver for the logger to receive messages and write them somewhere.
+pub type ChannelLogReceiver = crossbeam::channel::Receiver<AccessLogMessage>;
+
+/// https://github.com/tokio-rs/tracing/blob/master/tracing-appender/src/non_blocking.rs#L61-L70
+const DEFAULT_BUFFERED_LINES_LIMIT: usize = 128_000;
+
+/// Creates a new channel for access logs.
+pub fn create_log_channel() -> (ChannelLogSender, ChannelLogReceiver) {
```

Comment:
the `lossy_log` boolean flag should be included in the Sender type IMO, so have both:
```
create_lossy_log_channel()-> (ChannelLogSender, ChannelLogReceiver)
create_log_channel()-> (ChannelLogSender, ChannelLogReceiver)
```
with
```
struct ChannelLogSender {
  sender: crossbeam::channel::Sender<AccessLogMessage>
  lossy: bool
}
```

I don't find it great that this boolean flag is propagated at various places